test/torchaudio_unittest/datasets/utils_test.py

```diff
@@ -1,36 +1,40 @@
+import torch
 from torchaudio_unittest.common_utils import (
     TorchaudioTestCase,
-    get_asset_path,
+    TempDirMixin,
+    get_asset_path
 )
 
 from torchaudio.datasets import utils as dataset_utils
-from torchaudio.datasets.commonvoice import COMMONVOICE
 
-original_ext_audio = COMMONVOICE._ext_audio
 
+class Dataset(torch.utils.data.Dataset):
```

Comment:
The iterator protocol implemented here looks inconsistent between `__getitem__` and `__iter__`. i.e. index slicing will return `Tuple[Tensor, int]` while `__iter__` will return series of `int`s. Is `__iter__` method required?

I think something like the following would do
(I changed the returned Tensor to use the input argument `n`, and added variable `sample_rate` for the purpose of readability improvement.)

```python

def __getitem__(self, n):
    sample_rate = 8000
    waveform = n * torch.ones(2, 256)
    return waveform, sample_rate

def __len__(self) -> int:
    return 32
```

Reply:
What I noticed is that without `__iter__` method inside the `Dataset` class, the test will have in infinite loop [here](https://github.com/pytorch/audio/blob/71214b48548b1dcb6ebd581dd36a9d0e60af6837/torchaudio/datasets/utils.py#L258) and that's why we have timeout error in the CICD pipeline.

Reply:
Another option would also to use `IterableDataset`.

Reply:
> What I noticed is that without `__iter__` method inside the `Dataset` class, the test will have in infinite loop [here](https://github.com/pytorch/audio/blob/71214b48548b1dcb6ebd581dd36a9d0e60af6837/torchaudio/datasets/utils.py#L258) and that's why we have timeout error in the CICD pipeline.

I see. When you saw it, did the Dataset have `__len__` implementation? The fact that CommonVoice has `__getitem__` and `__len__` but not `__iter__` makes me think that `__len__` would cap the maximum iteration.

Reply:
Yes, `__len__` method was there when we had the infinite loop.
Without the `__len__` method the `test_disckcache_iterator` test will fail due to an error:
`>       self._cache: List = [None] * len(dataset)
E       TypeError: object of type 'Dataset' has no len()`

Reply:
I see. In that case, reusing the implementation of `__len__` and `__getitem__` in `__iter__` will give a consistent behavior.

```python
def __iter__(self):
    for i in range(len(self)):
        yield self[i]
```